base/math.jl

```diff
@@ -579,12 +579,18 @@ function hypot(x::T,y::T) where T<:AbstractFloat
         scale = one(scale)
     end
     h = sqrt(muladd(ax,ax,ay*ay))
-    if h <= 2*ay
-        delta = h-ay
-        h -= muladd(delta,delta-2*(ax-ay),ax*(2*delta - ax))/(2*h)
+    if Base.Math.FMA_NATIVE
+        hsquared = h*h
+        axsquared = ax*ax
+        h -= (fma(-ay,ay,hsquared-axsquared) + fma(h,h,-hsquared) - fma(ax,ax,-axsquared))/(2*h)
```

Comment:
Could also just use `muladd` here, since it will use `fma` internally if `Base.Math.FMA_NATIVE` is true.

Reply:
Explicitly calling `fma` here seems better: the branch is actually correct with `fma`, regardless of whether we're running on an FMA-native system or not, it's just that the performance will be bad on a non-FMA-native system.

Reply:
Particularly true in light of the fact that in the work leading up to my original merge on hypot we discovered that not all muladds were being converted to fma. That is why I explicitly used fma here.